core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -403,7 +403,9 @@ private SqlLibraryOperators() {
   @LibraryOperator(libraries = {BIG_QUERY})
   public static final SqlFunction DATE =
       new SqlFunction("DATE", SqlKind.OTHER_FUNCTION,
-          ReturnTypes.DATE_NULLABLE, null, OperandTypes.STRING,
+          ReturnTypes.DATE_NULLABLE, null,
+          OperandTypes.or(OperandTypes.DATE, OperandTypes.TIMESTAMP,
+              OperandTypes.DATETIME, OperandTypes.STRING),
```

Comment:
DATETIME includes DATE and TIMESTAMP.

Reply:
Yes have changed the OperandTypes to OperandTypes.DATETIME, OperandTypes.STRING